packages/contracts/test/verifier.test.ts

```diff
@@ -479,61 +565,84 @@ describe('Epoch key proof verifier', function () {
+                    state_tree_elements: new Array(STATE_TREE_DEPTH).fill(0),
+                    state_tree_indexes: new Array(STATE_TREE_DEPTH).fill(0),
+                    identity_secret: id.secret,
+                    data: Array(FIELD_COUNT).fill(0),
+                    sig_data: data,
+                    epoch,
+                    nonce,
+                    attester_id: owner.address,
+                    reveal_nonce: 0,
+                })
+            )
 
-        const data = 0
-        const nonce = 0
-        const r = await defaultProver.genProofAndPublicSignals(
-            Circuit.epochKey,
-            stringifyBigInts({
-                state_tree_elements: new Array(STATE_TREE_DEPTH).fill(0),
-                state_tree_indexes: new Array(STATE_TREE_DEPTH).fill(0),
-                identity_secret: id.secret,
-                data: Array(FIELD_COUNT).fill(0),
-                sig_data: data,
-                epoch,
-                nonce,
-                attester_id: attester.address,
-                reveal_nonce: 0,
-            })
-        )
+            const { publicSignals, proof } = new EpochKeyProof(
+                r.publicSignals,
+                r.proof
+            )
+            await expect(
+                epochKeyProofVerifier
+                    .connect(owner)
+                    .verifyAndCheckCaller(publicSignals, proof)
+            ).to.not.be.reverted
+        }
+        {
+            const data = 0
+            const nonce = 0
+            const randomAddress = BigInt(10)
+            const r = await defaultProver.genProofAndPublicSignals(
+                Circuit.epochKey,
+                stringifyBigInts({
+                    state_tree_elements: new Array(STATE_TREE_DEPTH).fill(0),
+                    state_tree_indexes: new Array(STATE_TREE_DEPTH).fill(0),
+                    identity_secret: id.secret,
+                    data: Array(FIELD_COUNT).fill(0),
+                    sig_data: data,
+                    epoch,
+                    nonce,
+                    attester_id: randomAddress,
+                    reveal_nonce: 0,
+                })
+            )
 
-        const v = await defaultProver.verifyProof(
-            Circuit.epochKey,
-            r.publicSignals,
-            r.proof
-        )
-        expect(v).to.be.true
-        const { publicSignals, proof } = new EpochKeyProof(
-            r.publicSignals,
-            r.proof
-        )
-        await expect(
-            unirepContract.verifyEpochKeyProof(publicSignals, proof)
-        ).to.be.revertedWithCustomError(unirepContract, 'InvalidStateTreeRoot')
+            const { publicSignals, proof } = new EpochKeyProof(
+                r.publicSignals,
+                r.proof
+            )
+            await expect(
+                epochKeyProofVerifier
+                    .connect(owner)
+                    .verifyAndCheckCaller(publicSignals, proof)
+            ).to.be.reverted
```

Comment:
When you check that a call is reverted you should verify that it's reverted with the correct error. See [here](https://github.com/Unirep/Unirep/blob/main/packages/contracts/test/attestation.test.ts#L48) for an example.

Reply:
Done, checking for revert message